ouroboros-consensus-test/test-consensus/Test/Consensus/Mempool/StateMachine.hs

```diff
@@ -0,0 +1,810 @@
+  , LedgerSupportsProtocol blk
+  , UnTick blk
+  , IOLike m
+  , NoThunks (Mempool m blk)
+  )
+  => LedgerConfig blk
+  -> LedgerState blk ValuesMK
+  -> (Int -> LedgerState blk ValuesMK -> Gen [GenTx blk])
+  -> m (StateMachine (Model blk) (Command blk) m (Response blk))
+sm cfg initialState gTxs = do
+  ior <- newTVarIO =<< mkSUT cfg initialState
+  pure $ StateMachine {
+      QC.initModel     = initModel cfg initialState
+    , QC.transition    = transitions cfg
+    , QC.precondition  = noPreconditions
+    , QC.postcondition = postconditions
+    , QC.invariant     = Nothing
+    , QC.generator     = generator gTxs
+    , QC.shrinker      = shrinker
+    , QC.semantics     = \c -> semantics c ior
+    , QC.mock          = mock
+    , QC.cleanup       = noCleanup
+    }
+
+smUnused ::
+  ( HasTxId (GenTx blk)
+  , LedgerSupportsMempool blk
+  , Eq (GenTx blk)
+  , Arbitrary (LedgerState blk ValuesMK)
+  , ToExpr (LedgerState blk ValuesMK)
+  , ToExpr (GenTx blk)
+  , Eq (TickedLedgerState blk ValuesMK)
+  , LedgerSupportsProtocol blk
+  , UnTick blk
+  )
+  => LedgerConfig blk
+  -> LedgerState blk ValuesMK
+  -> (Int -> LedgerState blk ValuesMK -> Gen [GenTx blk])
+  -> StateMachine (Model blk) (Command blk) IO (Response blk)
+smUnused cfg initialState gTxs = StateMachine {
+    QC.initModel     = initModel cfg initialState
+  , QC.transition    = transitions cfg
+  , QC.precondition  = noPreconditions
+  , QC.postcondition = postconditions
+  , QC.invariant     = Nothing
+  , QC.generator     = generator gTxs
+  , QC.shrinker      = shrinker
+  , QC.semantics     = undefined
+  , QC.mock          = mock
+  , QC.cleanup       = noCleanup
+  }
+
+{-------------------------------------------------------------------------------
+  Properties
+-------------------------------------------------------------------------------}
+
+prop_mempoolSequential ::
+  forall blk . ( HasTxId (GenTx blk)
+  , LedgerSupportsMempool blk
+  , Eq (GenTx blk)
+  , Arbitrary (LedgerState blk ValuesMK)
+  , ToExpr (LedgerState blk ValuesMK)
+  , ToExpr (GenTx blk)
+  , Eq (TickedLedgerState blk ValuesMK)
+  , LedgerSupportsProtocol blk
+  , ToExpr (TickedLedgerState blk ValuesMK)
+  , UnTick blk
+  , NoThunks (Mempool IO blk)
+  )
+  => LedgerConfig blk
+  -> LedgerState blk ValuesMK
+     -- ^ Initial state
+  -> (Int -> LedgerState blk ValuesMK -> Gen [GenTx blk])
+     -- ^ Transaction generator
+  -> Property
+prop_mempoolSequential cfg initialState gTxs = forAllCommands smUnused' Nothing $
+  \cmds -> monadicIO
+    (do
+        (hist, model, res) <- runCommandsWithSetup sm' cmds
+        prettyCommands smUnused' hist
+          $ checkCommandNames cmds
+          $ tabulate "Command sequence length"
+              [QC.lengthCommands cmds `bucketiseBy` 10]
+          $ tabulate "Maximum ticket number"
+              [(\(TicketNo t) -> t) (modelLastTicket model) `bucketiseBy` 5]
+          $ tabulate "Number of txs to add"
```

Comment:
We might want to add labels for actions with accepted, rejected, and unprocessed transactions.